app/utils/validation.py

```diff
@@ -39,6 +39,9 @@ def clean_and_validate_retention_period(retention_period):
 
 
 def validate_filename(filename):
+    if len(filename) > current_app.config["MAX_CUSTOM_FILENAME_LENGTH"]:
+        raise ValueError("`filename` cannot be longer than 100 characters")
```

Comment:
Optional: You could replace the hardcoded 100 with the config variable

Reply:
oops :)